saleor/checkout/tests/test_checkout_complete.py

```diff
@@ -923,3 +924,42 @@ def test_create_order_use_translations(
 
     assert order_line.translated_product_name == translated_product_name
     assert order_line.translated_variant_name == translated_variant_name
+
+
+def test_create_order_deactivates_unused_payments(
+    checkout_with_item, address, customer_user, payment_kwargs
+):
+    # given
+    checkout_with_item.shipping_address = address
+    checkout_with_item.save()
+    manager = get_plugins_manager()
+    lines = fetch_checkout_lines(checkout_with_item)
+    checkout_info = fetch_checkout_info(checkout_with_item, lines, [], manager)
+
+    unused_payment = Payment.objects.create(
+        **{
+            **payment_kwargs,
+            **{
+                "checkout": checkout_with_item,
+                "order": None,
+            },
+        }
+    )
+    payments_count = Payment.objects.count()
+    # when
```

Comment:
I will suggest adding empty lines before `# when` and `# then` comments to separate different parts.